pkg/policy/l4.go

```diff
@@ -153,10 +162,14 @@ type PerSelectorPolicy struct {
 
 	// Pre-computed HTTP rules with resolved k8s secrets
 	// Computed after rule merging is complete!
+	//
+	// +deepequal-gen=false
 	EnvoyHTTPRules *cilium.HttpNetworkPolicyRules `json:"-"`
 
 	// CanShortCircuit is true if all 'EnvoyHTTPRules' may be
 	// short-circuited by other matches.
+	//
+	// +deepequal-gen=false
 	CanShortCircuit bool `json:"-"`
```

Comment:
I feel like these fields *should* be included in the new `DeepEqual()`, but kept excluded from `Equal()`. One way of achieving this would be to keep them excluded from autogenerated `deepEqual()` and manually compare them in `DeepEqual()`?

Reply:
`EnvoyHTTPRules` is included in this structure because it needs to be computed while the context for secret manager is available. It has been excluded from `Equal` to allow unit tests to behave as if it did not exist. Might be better to amend the unit tests to include the Envoy translation of the rules, but on the other hand it would complicate the policy package unit tests with envoy package implementation details.